### ISSUE TITLE
AnnotationLocation check does not force method annotation to be exactly after javadoc.

### ISSUE THREAD

**Issue by pbaranchikov** [checkstyle/checkstyle#3213]
According to the check description, both annotations (Nullable and Nonnull) should be reported as violations.
- The first one is before javadoc.
- The second one is not exactly after javadoc (it has _protected_ keyword between javadoc and annocation).
```java
public class TestClass {
    @Nullable
    /**
     * Some javadoc here
     */
    protected @Nonnull static boolean doSomething() {
        System.out.print("Hello, world!");
    }
}
```

**Comment by rnveach** [comment]
@romani This does look like it should be a violation.

**Comment by Atharv3221** [comment]
https://checkstyle.org/checks/annotation/annotationlocation.html#Notes

**Comment by romani** [comment]
please use InvalidJavadocPosition  https://checkstyle.org/checks/javadoc/invalidjavadocposition.html 
```
$ cat Test.java
public class TestClass {
    @Nullable
    /**
     * Some javadoc here
     */
    protected @Nonnull static boolean doSomething() {
        System.out.print("Hello, world!");
    }
}

$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <module name="TreeWalker">
       <module name="InvalidJavadocPosition" />
    </module>
</module>

$ java -jar checkstyle-13.8.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/roman/tmp/Test.java:3:5: Javadoc comment is placed in the wrong location. [InvalidJavadocPosition]
Audit done.
Checkstyle ends with 1 errors.
```
